src/cdk/datetime/datepicker.ts

```diff
@@ -0,0 +1,131 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  Optional,
+  ViewEncapsulation,
+  OnDestroy,
+  ContentChild,
+} from '@angular/core';
+import {Subject, Subscription} from 'rxjs';
+import {DateAdapter} from './date-adapter';
+import {CdkDatepickerInput} from './datepicker-input';
+import {CalendarView} from './calendar-view';
+
+/** Used to generate a unique ID for each datepicker instance. */
+let datepickerUid = 0;
+
+// TODO(vwei): We use a component instead of a directive here so the user can use implicit
+// template reference variables (e.g. #d vs #d="cdkDatepicker"). We can change this to a directive
+// if angular adds support for `exportAs: '$implicit'` on directives.
+/** Component responsible for managing the datepicker CDK. */
+@Component({
+  moduleId: module.id,
+  selector: 'cdk-datepicker',
+  template: '',
```

Comment:
maybe for the cdk one it makes sense to just make the template `<ng-content></ng-content>`?

Reply:
Yes, I put it in previous PR but forgot to add in this one.

Reply:
In that case we can remove the TODO comment above about using a directive for this, directives don't support templates

Reply:
TODO comment removed